[Bug]: CI release fails due to missing .NET SDK setup

## Description

The release job fails when pushing git tags because `dotnet build` runs without the proper .NET SDK installed. The CI workflow needs to set up .NET 8 SDK in all jobs that run typecheck.

## Context

The CI workflow sets up Python for the Python fixtures but does **not** set up .NET for the C# fixtures. GitHub's `ubuntu-latest` now ships with .NET 10.0.101 (preview) by default, but our project targets `net8.0`.

When the changesets action pushes git tags during release:
1. Husky pre-push hook triggers (`pnpm typecheck`)
2. Turbo runs typecheck for `@pietgk/devac-fixtures-csharp`
3. `dotnet build` runs with .NET 10 preview SDK (wrong version)
4. Build fails with `error MSB4018: The "GenerateDepsFile" task failed unexpectedly`
5. Git push fails, release job fails

**Failed run:** https://github.com/pietgk/vivief/actions/runs/20335661044/job/58421625411

## Acceptance Criteria

- [ ] Add `actions/setup-dotnet@v4` with pinned .NET 8.x SDK to the `build` job
- [ ] Add `actions/setup-dotnet@v4` with pinned .NET 8.x SDK to the `release` job
- [ ] C# fixtures typecheck passes in all CI jobs
- [ ] Release job successfully completes (tags push works)

## Affected Packages

- [ ] devac-core
- [ ] devac-cli
- [ ] devac-mcp
- [x] fixtures-csharp
- [ ] Documentation
- [x] CI/CD

## Constraints

- Changeset required: No (CI/infrastructure change)
- ADR required: No
- Breaking change: No

## Open Questions

None - the fix is straightforward: add .NET SDK setup with a pinned version.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,6 +22,11 @@ jobs:
         with:
           python-version: "3.11"
 
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: "8.0.x"
+
       - name: Setup pnpm
         uses: pnpm/action-setup@v4
 
@@ -117,6 +122,16 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: "8.0.x"
 
       - name: Setup pnpm
         uses: pnpm/action-setup@v4

